data_steward/cdr_cleaner/cleaning_rules/create_person_ext_table.py

```diff
@@ -0,0 +1,129 @@
+LEFT JOIN  `{{project}}.{{dataset}}.observation_ext` e
+ON o.observation_id = e.observation_id AND o.observation_source_concept_id = 1585249
+""")
+
+tables = ['person_ext']
+
+
+class CreatePersonExtTable(BaseCleaningRule):
+    """
+    Create person_ext table after DEID, adds three non-standard fields:
+    state_of_residence_concept_id, state_of_residence_source_value
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = ('Create person_ext table')
+        super().__init__(issue_numbers=['DC1012'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.DEID_CLEAN],
+                         affected_tables=tables,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+
+        query_list = []
+        for table in tables:
+            query_list.append({
+                cdr_consts.QUERY:
+                    PERSON_EXT_TABLE_QUERY.render(project=self.project_id,
+                                                  dataset=self.dataset_id),
+                cdr_consts.DESTINATION_TABLE:
+                    table,
+                cdr_consts.DESTINATION_DATASET:
+                    self.dataset_id,
+                cdr_consts.DISPOSITION:
+                    WRITE_TRUNCATE
+            })
+
+        return query_list
+
+    def setup_rule(self, client):
+        """
+        Function to run any data upload options before executing a query.
+        """
+        pass
+
+    def setup_validation(self, client):
+        """
+        Run required steps for validation setup
+        """
+        raise NotImplementedError("Please fix me.")
+
+    def validate_rule(self, client):
+        """
+        Validates the cleaning rule which deletes or updates the data from the tables
+        """
+        raise NotImplementedError("Please fix me.")
+
+    def get_sandbox_tablenames(self):
+        sandbox_table_names = list()
+        for i in range(0, len(self._affected_tables)):
+            sandbox_table_names.append(self._issue_numbers[0].lower() + '_' +
+                                       self._affected_tables[i])
+        return sandbox_table_names
```

Comment:
This isn't actually accurate.  This rule doesn't make use of sandbox tables.  This should actually return an empty list.

Reply:
I had a feeling, I tried taking this entire function out and realized it broke the class, so I left as is, but I will have this return an empty list instead, thank you!